spec/secrets_dsl_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'spec_helper'
+require 'vcr_helper'
+
+RSpec.describe 'secrets DSL', vcr: VCR_OPTS do
+
+  let(:region) { 'us-east-2' }
+  let(:env_name) { 'openstax-aws-ruby-secrets' }
+  let(:namespace) { 'secrets-dsl-spec/main' }
+
+  before(:each) {
+    @logger = spy("logger")
+
+    OpenStax::Aws.configure do |config|
+      config.cfn_template_bucket_name = "openstax-sandbox-cfn-templates"
+      config.cfn_template_bucket_region = "us-west-2"
+      config.stack_waiter_delay = vcr_recording? ? 5 : 0
+      config.logger = vcr_recording? ? Logger.new(STDERR) : @logger
+    end
+
+    allow_any_instance_of(OpenStax::Aws::Template).to receive(:s3_folder) { "spec-templates" }
+  }
+
+  it 'creates secrets when stack created and deletes when deleted' do
+    deployment = SecretsDslSpec::Deployment.new(env_name: env_name, region: region, dry_run: false)
+    deployment.create(some_sha: "1b2ebfd91dd9fb34d58c834cbb70a21c6479ba8e", bucket_name: "aws-ruby-spec-secrets-dsl-create")
+
+    expect_secrets_in_parameter_store({
+      "random" => /[a-z0-9]{4}/,
+      "interpolated" => "https://something-deployment-specific",
+      "for_shared_substitution" => "howdy y'all",
+      "content_key" => "something-deployment-specific",
+      "more_secrets" => "foo"
+    })
+
+    deployment.delete
+
+    expect(secrets.data).to be_empty
+  end
+
+  it 'updates secrets when stack updated' do
```

Comment:
kinda feels like these tests are testing too much (test focus too broad).  one give away is the does this AND does that in the it description.  

imo, all rails devs should read thru http://www.betterspecs.org/.  there are many suggestions in here that this spec could improve on imo (one is long test descriptions, one is single test expectations).

Reply:
Yeah I got lazy.  May not fix this one.  I did add some more lower-level tests.  The real fix would make it easier to test these without a lot of set up (right now requires separate Ruby class and CloudFormation template)